engineering-plans/0001-graphql-query-prefetching.md

```diff
@@ -196,24 +209,30 @@ select c.*, p.id as parent_id
  order by c.{sort_key}
 ```
 
-For this type,
+##### Single child per parent
+
 ```sql
-expand_parents  := unnest({parent_ids}) as p(id)
-linked_children := p.id = c.{parent_field}
+select c.*, c.{parent_field} as parent_id
+  from children c
+ where c.{parent_field} = any({parent_ids})
+   and .. other conditions on c ..
+ limit {parent_ids.len} + 1
```

Comment:
I couldn't find any big cases like this in the wild, the only I found dealt with a very small amount of data. So I guess we don't need to focus too much on it at this point, but imo it would be more consistent with what we do in the other single child optimizations to join with the unnested ids, something like:
```
select c.*, c.{parent_field} as parent_id
  from unnest({parent_ids}) as p(id), children c
 where c.{parent_field} = p.id
   and .. other conditions on c ..
 limit {parent_ids.len} + 1
```